components/browser/awesomebar/src/main/java/mozilla/components/browser/awesomebar/BrowserAwesomeBar.kt

```diff
@@ -154,7 +156,24 @@ class BrowserAwesomeBar @JvmOverloads constructor(
 
             providers.forEach { provider ->
                 launch {
+                    // At this point, we have a timing value for a provider.
+                    // We have a choice here - we can try grouping different timings together for a
+                    // single user input value, or we can just treat them as entirely independent values.
+                    // These timings are correlated with each other in a sense that they act on the same
+                    // inputs, and are executed at the same time. However, our goal here is to track performance
+                    // of the providers. Each provider acts independently from another; recording their performance
+                    // at an individual level will allow us to track that performance over time.
+                    // Tracked value will be reflected both in perceived user experience (how quickly results from
+                    // a provider show up),
```

Comment:
nit: looks like part of the sentence got lost: both in perceived user experience.., and...